pkg/reconciler/shared/tektonconfig/upgrade/post_upgrade.go

```diff
@@ -70,3 +80,66 @@ func upgradeStorageVersion(ctx context.Context, logger *zap.SugaredLogger, k8sCl
 
 	return nil
 }
+
+// Remove the all  cluster tasks installer sets
+func removeAllClusterTasksInstallerSets(ctx context.Context, logger *zap.SugaredLogger, k8sClient kubernetes.Interface, operatorClient versioned.Interface, restConfig *rest.Config) error {
+
+	if !v1alpha1.IsOpenShiftPlatform() {
+		return nil
+	}
+
+	tisClient := operatorClient.OperatorV1alpha1().TektonInstallerSets()
+
+	ctLS := v1.LabelSelector{
+		MatchLabels: map[string]string{
+			v1alpha1.InstallerSetType: installerSetType(customType, "ClusterTask"),
+		},
+	}
+	ctLabel, err := common.LabelSelector(ctLS)
+	if err != nil {
+		return err
+	}
+	if err := tisClient.DeleteCollection(ctx, v1.DeleteOptions{}, v1.ListOptions{
```

Comment:
the same code for deleting the different installersetType
Why not just simple loop over []string{"ClusterTask", "CommunityClusterTask", "VersionedClusterTask"}

Reply:
yes, we can do and I try once